tenant-partitioning-config-service-impl/src/main/java/org/hypertrace/tenantpartitioning/config/service/TenantPartitioningConfigServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.tenantpartitioning.config.service.store.TenantPartitionConfigStore;
+import org.hypertrace.tenantpartitioning.config.service.store.TenantPartitionGroupConfigKey;
+import org.hypertrace.tenantpartitioning.config.service.v1.*;
+
+@Slf4j
+public class TenantPartitioningConfigServiceImpl
+    extends TenantPartitioningConfigServiceGrpc.TenantPartitioningConfigServiceImplBase {
+
+  public static final String GENERIC_CONFIG_SERVICE = "generic.config.service";
+
+  private final TenantPartitionConfigStore tenantIsolationConfigStore;
+  private final TenantPartitioningConfigServiceRequestValidator validator;
+
+  @Inject
+  public TenantPartitioningConfigServiceImpl(
+      Config config,
+      TenantPartitionConfigStore tenantIsolationConfigStore,
+      TenantPartitioningConfigServiceRequestValidator validator) {
+    this.tenantIsolationConfigStore = tenantIsolationConfigStore;
+    this.validator = validator;
+    Config cfg = config.getConfig(GENERIC_CONFIG_SERVICE);
+    // this.tenantIsolationConfigStore = TenantPartitionConfigStoreProvider.getDocumentStore(cfg);
+  }
+
+  public void getTenantPartitionGroupConfigs(
+      GetTenantPartitionGroupConfigsRequest request,
+      StreamObserver<GetTenantPartitionGroupConfigsResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      List<TenantPartitionGroupConfig> configDTOs = tenantIsolationConfigStore.getAllGroupConfigs();
+      List<TenantPartitionGroupConfig> configs =
+          configDTOs.stream()
+              .map(
+                  TenantPartitionGroupConfigDTO ->
+                      TenantPartitionGroupConfig.newBuilder()
+                          .setGroupName(TenantPartitionGroupConfigDTO.getGroupName())
+                          .setWeight(TenantPartitionGroupConfigDTO.getWeight())
+                          .addAllMemberTenantIds(
+                              TenantPartitionGroupConfigDTO.getMemberTenantIdsList())
+                          .build())
+              .collect(Collectors.toList());
+
+      responseObserver.onNext(
+          GetTenantPartitionGroupConfigsResponse.newBuilder().addAllConfigs(configs).build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("getAllTenantPartitionGroupConfigs failed ", e);
+      responseObserver.onError(e);
+    }
+  }
+
+  public void createTenantPartitionGroupConfig(
+      CreateTenantPartitionGroupConfigRequest request,
+      StreamObserver<CreateTenantPartitionGroupConfigResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      TenantPartitionGroupConfig upsertedConfigDTO =
+          tenantIsolationConfigStore.upsert(
+              TenantPartitionGroupConfig.newBuilder()
+                  .setId(
```

Comment:
If we're not going to use uuids for identification, make sure to validate that the ID handed to create doesn't already exist.

For that matter, probably need to validate (on both create and update) that the tenant ids provided do not belong to another group.